### PR TITLE
chore(ci): upgrade gha

### PR DESCRIPTION
**Pull request by sc979**
## Description

upgrade GHA
fix backend-lint deprecated requirement

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

### CODE REVIEW

File: .github/actions/backend-dependencies/action.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ runs:
       shell: bash
 
     - name: Cache vendor directory
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/backend-dependencies/action.yml:27]:
```suggestion
      uses: actions/cache/save@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/actions/frontend-build/action.yml [hunk 2]
```diff
@@ -54,13 +54,13 @@ runs:
 
     - name: Cache index.html file
       if: ${{ inputs.index_cache_key != '' && inputs.index_file != '' }}
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/frontend-build/action.yml:57]:
```suggestion
      uses: actions/cache/save@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/actions/frontend-build/action.yml [hunk 3]
```diff
@@ -54,13 +54,13 @@ runs:
...
       with:
         path: ${{ inputs.index_file }}
         key: ${{ inputs.index_cache_key }}
 
     - name: Cache static directory
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/frontend-build/action.yml:63]:
```suggestion
      uses: actions/cache/save@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/actions/package/action.yml [hunk 4]
```diff
@@ -57,23 +57,23 @@ runs:
   steps:
     - name: Restore index file cache
       if: "${{ inputs.frontend_index_file != '' && inputs.frontend_index_cache_key != '' }}"
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/package/action.yml:60]:
```suggestion
      uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/actions/package/action.yml [hunk 5]
```diff
@@ -57,23 +57,23 @@ runs:
...
       with:
         path: ${{ inputs.frontend_index_file }}
         key: ${{ inputs.frontend_index_cache_key }}
         fail-on-cache-miss: true
 
     - name: Restore static directory cache
       if: "${{ inputs.frontend_static_directory != '' && inputs.frontend_static_cache_key != '' }}"
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/package/action.yml:68]:
```suggestion
      uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/actions/package/action.yml [hunk 6]
```diff
@@ -57,23 +57,23 @@ runs:
...
       with:
         path: ${{ inputs.frontend_static_directory }}
         key: ${{ inputs.frontend_static_cache_key }}
         fail-on-cache-miss: true
 
     - name: Restore widgets directory cache
       if: "${{ inputs.frontend_widgets_directory != '' && inputs.frontend_widgets_cache_key != '' }}"
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/package/action.yml:76]:
```suggestion
      uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/actions/package/action.yml [hunk 7]
```diff
@@ -87,15 +87,15 @@ runs:
 
     - name: Restore vendor directory cache
       if: "${{ inputs.backend_vendor_directory != '' && inputs.backend_vendor_cache_key != '' }}"
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/package/action.yml:90]:
```suggestion
      uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/actions/package/action.yml [hunk 8]
```diff
@@ -87,15 +87,15 @@ runs:
...
       with:
         path: ${{ inputs.backend_vendor_directory }}
         key: ${{ inputs.backend_vendor_cache_key }}
         fail-on-cache-miss: true
 
     - name: Restore translation directory cache
       if: "${{ inputs.translation_directory != '' && inputs.translation_cache_key != '' }}"
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/package/action.yml:98]:
```suggestion
      uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/actions/package/action.yml [hunk 9]
```diff
@@ -173,7 +173,7 @@ runs:
         retention-days: 1
 
     - name: Cache packaged files
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/package/action.yml:176]:
```suggestion
      uses: actions/cache/save@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/actions/release-sources/action.yml [hunk 10]
```diff
@@ -48,31 +48,31 @@ runs:
 
     - name: Restore index file cache
       if: "${{ inputs.frontend_index_file != '' && inputs.frontend_index_cache_key != '' }}"
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/release-sources/action.yml:51]:
```suggestion
      uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/actions/release-sources/action.yml [hunk 11]
```diff
@@ -48,31 +48,31 @@ runs:
...
       with:
         path: ${{ inputs.frontend_index_file }}
         key: ${{ inputs.frontend_index_cache_key }}
         fail-on-cache-miss: true
 
     - name: Restore static directory cache
       if: "${{ inputs.frontend_static_directory != '' && inputs.frontend_static_cache_key != '' }}"
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/release-sources/action.yml:59]:
```suggestion
      uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/actions/release-sources/action.yml [hunk 12]
```diff
@@ -48,31 +48,31 @@ runs:
...
       with:
         path: ${{ inputs.frontend_static_directory }}
         key: ${{ inputs.frontend_static_cache_key }}
         fail-on-cache-miss: true
 
     - name: Restore vendor directory cache
       if: "${{ inputs.backend_vendor_directory != '' && inputs.backend_vendor_cache_key != '' }}"
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/release-sources/action.yml:67]:
```suggestion
      uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/actions/release-sources/action.yml [hunk 13]
```diff
@@ -48,31 +48,31 @@ runs:
...
       with:
         path: ${{ inputs.backend_vendor_directory }}
         key: ${{ inputs.backend_vendor_cache_key }}
         fail-on-cache-miss: true
 
     - name: Restore translation directory cache
       if: "${{ inputs.translation_directory != '' && inputs.translation_cache_key != '' }}"
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/actions/release-sources/action.yml:75]:
```suggestion
      uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/workflows/cypress-e2e-parallelization.yml [hunk 14]
```diff
@@ -84,7 +84,7 @@ jobs:
 
       - name: Restore packages
         if: "${{ inputs.package_cache_key != '' && inputs.package_directory != '' && contains(matrix.feature, 'platform-') }}"
-        uses: actions/cache@e12d46a63a90f2fae62d114769bbf2a179198b5c # v3.3.3
+        uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/workflows/cypress-e2e-parallelization.yml:87]:
```suggestion
        uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/workflows/package.yml [hunk 15]
```diff
@@ -117,7 +117,7 @@ jobs:
           cache_key: unsigned-${{ inputs.cache_key }}
 
       - if: ${{ inputs.package_extension == 'deb' }}
-        uses: actions/cache@e12d46a63a90f2fae62d114769bbf2a179198b5c # v3.3.3
+        uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/workflows/package.yml:120]:
```suggestion
        uses: actions/cache/save@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/workflows/package.yml [hunk 16]
```diff
@@ -141,7 +141,7 @@ jobs:
 
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
 
-      - uses: actions/cache@e12d46a63a90f2fae62d114769bbf2a179198b5c # v3.3.3
+      - uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/workflows/package.yml:144]:
```suggestion
      - uses: actions/cache/save@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/workflows/package.yml [hunk 17]
```diff
@@ -152,7 +152,7 @@ jobs:
       - run: rpmsign --addsign ./*.${{ inputs.package_extension }}
         shell: bash
 
-      - uses: actions/cache@e12d46a63a90f2fae62d114769bbf2a179198b5c # v3.3.3
+      - uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/workflows/package.yml:155]:
```suggestion
      - uses: actions/cache/save@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/workflows/web.yml [hunk 18]
```diff
@@ -196,7 +196,7 @@ jobs:
           done
         shell: bash
 
-      - uses: actions/cache@e12d46a63a90f2fae62d114769bbf2a179198b5c # v3.3.3
+      - uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/workflows/web.yml:199]:
```suggestion
      - uses: actions/cache/save@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

**Comment by sc979** [comment]
tested in job : https://github.com/centreon/centreon/pull/3234/checks?sha=96c13812f37d78ac5f19b2b4a82a2e0fb1863b48

**Review by kduret — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**